_posts/2020-04-02-whats-the-worst-that-can-happen.md

```diff
@@ -0,0 +1,55 @@
+- Environment: Production
+- Search: All our Java Application Servers
+- Command: Restart Java Application
+- Concurrency: 1
+
+These exact steps had worked in all our other environments without issue.
+
+## What Went Wrong?
+
+As we all know, mistakes happen, people are imperfect beings, that is why we script as much of our work as we can. In this case the Engineer forgot to set the Concurrency on the job to restart the applications. In the Application Restart job, this would have caused a problem, but not a major one, as it always starts with a single server, and prompts the Engineer to check that the application has started cleanly before moving on. But that job was not used, the job that was used was an older job, it had fewer safe guards, and unfortunately it had a default value for the concurrency.
+![](/images/jenkins_concurrency.png)
+As this job is used widely within the business, and may be used across hundreds of servers at a time it makes sense to have a high default concurrency, however in this instance that was sufficient to restart the application on all the java application servers at once. These applications handle a number of functions surrounding customer logins, consequently on the 23rd of March, at 10:27am all Customer logins across all Sky Betting and Gaming products failed, for approximately one minute.
+
+## What Went Well
+
+We try to work with a no blame culture. It isn't the fault of the individual, it is the fault of the platforms and procedures that allowed a mistake to have an undesired impact. So when the Engineer performing this change immediately put their hand up and said "I have caused an outage" it allowed the on-call engineers, and other interested parties to get on with the task of fixing the problem, and investigating the impact, without having to look at why the incident happened.
+
+It's important not to panic when things go wrong. A natural instinct is to "Do Some Thing!" when things are going wrong, but the wrong thing done quickly can sometimes do more damage than doing nothing. Based on the Engineer's account of what happened during this incident they saw the restart job connect to all the java application servers at once, and nearly killed the job then and there. This would have stopped that job from bringing the applications back up, and would have not been good. Fortunately they paused, and thought about what they wanted to do, and didn't take that action. This is always a useful way to handle the panic you will feel in that situation, pause and reflect on what you want to do, the extra time can bring clarity, it is that clarity that panic robs from us.
```

Comment:
```suggestion
It's important not to panic when things go wrong. A natural instinct is to "Do Some Thing!" when things are going wrong, but the wrong thing done quickly can sometimes do more damage than doing nothing. Based on the Engineer's account of what happened during this incident they saw the restart job connect to all the Java application servers at once, and nearly killed the job then and there. This would have stopped that job from bringing the applications back up, and would have not been good. Fortunately they paused, and thought about what they wanted to do, and didn't take that action. This is always a useful way to handle the panic you will feel in that situation, pause and reflect on what you want to do, the extra time can bring clarity, it is that clarity that panic robs from us.
```